src/chain/mod.rs

```diff
@@ -1144,15 +1491,29 @@ impl ChainSource {
 }
 
 impl Filter for ChainSource {
-	fn register_tx(&self, txid: &bitcoin::Txid, script_pubkey: &bitcoin::Script) {
+	fn register_tx(&self, txid: &Txid, script_pubkey: &Script) {
 		match self {
 			Self::Esplora { tx_sync, .. } => tx_sync.register_tx(txid, script_pubkey),
+			Self::Electrum { electrum_runtime_client, registered_txs_cache, .. } => {
+				match electrum_runtime_client.read().unwrap().as_ref() {
+					Some(client) => client.register_tx(txid, script_pubkey),
+					None => {
+						registered_txs_cache.lock().unwrap().push((*txid, script_pubkey.to_owned()))
```

Comment:
Is it fully safe with the two locks?

Reply:
Just asking the question to be sure. I think in `start` they are taken in the same order, so then it should be good?

Reply:
Checked in more detail, and indeed seems safe. It would be better perhaps if this kind of analysis wasn't necessary. Nested locks are a risk in general I think. But not sure if there is a way to do it in rust.

Reply:
How about using just one lock, locking a struct containing `electrum_runtime_client` and the pending txs/outs?

Reply:
Yes, this looks great! Also nice use of the enum that doesn't even contain pending txes if it is started. Educational for me!